### ISSUE TITLE
[RFE] Save environment settings in config.json

### ISSUE THREAD

**Issue by dariodsa** [mattermost/mattermost-server#17294]
#### Summary
According to the [documentation](https://docs.mattermost.com/administration/config-settings.html), Mattermost can be configured using evironment variables. It would be nice if those changes that are coming from the environment variables would be written to the ``config.json`` so that all configuration is at one place. 

#### Steps to reproduce
Mattermost 5.32
1. create environment variable ``MM_SERVICESETTINGS_SITEURL`` with value ``http://mattermost.test.local``
2. examine config file (``config.json``) after Mattermost is loaded
3. SiteURL hasn't changed 

#### Expected behavior
Mattermost should save current settings into it's configuration file, so source of the truth is only one file. 

#### Observed behaviour (that appears unintentional) 
```
~/config $ cat config.json | grep -i siteurl
        "SiteURL": "",
~/config $ echo $MM_SERVICESETTINGS_SITEURL
http://mattermost.test.local
~/config $ mattermost config get ServiceSettings.SiteURL
ServiceSettings.SiteURL: "http://mattermost.test.local"
```

#### Possible fixes
During loading phase, values from the environment variables should be written in the ``config.json``, overwriting the existing one.  After that is set, we can start loading configuration from the ``config.json`` so that file can become source of the truth. 
If you agree with me, I would like to take this issue and start working on it.

**Comment by amyblais** [comment]
@grundleborg / @agnivade Is this something we'd accept a PR for? (or should I ping another team to take a look)

**Comment by agnivade** [comment]
Hello @dariodsa - The observed behavior is indeed intentional and correct.

Environment variables are meant to be an override for the values in config.json. The way it works is:
- First the config file is loaded.
- The parameters passed from CLI are overridden. (We don't have any CLI parameters that map to config.json values currently).
- Then parameters from environment variables are overridden.

This sort of mechanism is a standard in many deployments and allows the operator to override various things the way they prefer. We don't want to overwrite the config.json values because the base values would be lost, and then we lose the significance of environment variables.

**Comment by dariodsa** [comment]
Ok, I can understand your reasons, plus ``Mattermost`` says that in the ``Admin console`` when someone tries to edit something that comes from an environment variable. 
But, it would be a nice feature for users that are working with ``Mattermost`` in the containerized environment to see which environment variables are loaded from the ``Mattermost`` side at the beginning in logs. For example with log level ``INFO`` we could print a message like:
```
Mattermost is overriding the ServiceSettings.SiteURL value from config.json with the value of environment variable MM_SERVICESETTINGS_SITEURL.
```
I hope that my request sounds reasonable.

**Comment by agnivade** [comment]
It would be great if you can explain the context behind this. What exactly are you trying to achieve by parsing the logs?

If you want to know what are the things that are being overridden, you already have the environment variables. If you want the final config, you can get that using the CLI.

**Comment by dariodsa** [comment]
I don't want to parse anything, I just want from logs to see that Mattermost will actually use those environment variables. For example if I misspell  some environment variable name, I would be able to see that from logs because Mattermost wouldn't use it. In case of running Mattermost in containerized environment, environment variables are most important and easiest way to alter Mattermost's settings. That why I think that those messages in Logs would be more than appreciated.

Dario

**Comment by agnivade** [comment]
Any reason you want to see that from only the logs and not in some other way? You can use this [API](https://api.mattermost.com/#tag/system/paths/~1config~1environment/get) to know all the config parameters that have been overridden.

**Comment by dariodsa** [comment]
I don't see any reason why Mattermost wouldn't put that information in log file. That is a worth information during Mattermost boot-up process. If it bothers you, I don't mind setting log-level of that information to ``DEBUG``. 
Of course, I can later see which variables are overridden in many ways much later, but let's take one example. Imagine I am setting up database's settings in environment variables (driver name and driver settings). If that is configured in the wrong way, Mattermost wouldn't boot up. In that case, I would like to see that Mattermost is really taking values of environments variables to make my debug process easier. In that case, I would see that Mattermost is using them and with which value.

**Comment by agnivade** [comment]
Yes, I think that's reasonable. I would be okay with showing what config settings are being overridden, at a `DEBUG` level. But we cannot show the value of that variable in the logs because they might leak sensitive information.

**Comment by dariodsa** [comment]
That makes sense. 
Thanks

**Comment by RahulJOD** [comment]
Hi, Please assign this issue to me. I want to contribute.
